File: py-opsml/src/scouter/alert.rs
use pyo3::prelude::*;
use scouter_client::*;

pub fn add_alert_module(m: &Bound<'_, PyModule>) -> PyResult<()> {
    m.add_class::<SpcAlertType>()?;
    m.add_class::<AlertZone>()?;
    m.add_class::<SpcAlert>()?;
    m.add_class::<SpcAlertRule>()?;
    m.add_class::<SpcAlertConfig>()?;
    m.add_class::<PsiAlertConfig>()?;
    m.add_class::<PsiNormalThreshold>()?;
    m.add_class::<PsiChiSquareThreshold>()?;
    m.add_class::<PsiFixedThreshold>()?;
    m.add_class::<AlertThreshold>()?;
    m.add_class::<AlertCondition>()?;
    m.add_class::<CustomMetricAlertConfig>()?;
    m.add_class::<AlertCondition>()?;
    m.add_class::<GenAIAlertConfig>()?;
    m.add_class::<SlackDispatchConfig>()?;
    m.add_class::<OpsGenieDispatchConfig>()?;
    m.add_class::<ConsoleDispatchConfig>()?;
    m.add_class::<AlertDispatchType>()?;
    Ok(())
}

File: py-opsml/python/opsml/scouter/alert.py
# type: ignore
# pylint: disable=no-name-in-module
from . import alert as _alert_impl

OpsGenieDispatchConfig = _alert_impl.OpsGenieDispatchConfig
ConsoleDispatchConfig = _alert_impl.ConsoleDispatchConfig
AlertDispatchType = _alert_impl.AlertDispatchType
AlertThreshold = _alert_impl.AlertThreshold
AlertZone = _alert_impl.AlertZone
CustomMetricAlertCondition = _alert_impl.CustomMetricAlertCondition
CustomMetricAlertConfig = _alert_impl.CustomMetricAlertConfig
PsiAlertConfig = _alert_impl.PsiAlertConfig
SpcAlert = _alert_impl.SpcAlert
SpcAlertConfig = _alert_impl.SpcAlertConfig
SpcAlertRule = _alert_impl.SpcAlertRule
SpcAlertType = _alert_impl.SpcAlertType
SlackDispatchConfig = _alert_impl.SlackDispatchConfig
PsiNormalThreshold = _alert_impl.PsiNormalThreshold
PsiChiSquareThreshold = _alert_impl.PsiChiSquareThreshold
PsiFixedThreshold = _alert_impl.PsiFixedThreshold
LLMMetricAlertCondition = _alert_impl.LLMMetricAlertCondition
LLMAlertConfig = _alert_impl.LLMAlertConfig


__all__ = [
    "AlertZone",
    "SpcAlertType",
    "SpcAlertRule",
    "PsiAlertConfig",
    "PsiAlertConfig",
    "SpcAlertConfig",
    "SpcAlert",
    "AlertThreshold",
    "CustomMetricAlertCondition",
    "CustomMetricAlertConfig",
    "SlackDispatchConfig",
    "OpsGenieDispatchConfig",
    "ConsoleDispatchConfig",
    "AlertDispatchType",
    "PsiNormalThreshold",
    "PsiChiSquareThreshold",
    "PsiFixedThreshold",
    "LLMMetricAlertCondition",
    "LLMAlertConfig",
]
